r/R/csv.R

```diff
@@ -623,8 +632,38 @@ readr_to_csv_convert_options <- function(na,
 #' @include arrow-package.R
 write_csv_arrow <- function(x,
                             sink,
+                            na = NULL,
+                            append = NULL,
+                            quote = NULL,
+                            escape = NULL,
+                            eol = NULL,
+                            num_threads = NULL,
+                            progress = NULL,
                             include_header = TRUE,
                             batch_size = 1024L) {
+
+  passed_args <- as.list(match.call()[-1])
+
+  arrow_write_opts <- names(formals(CsvWriteOptions$create))
+  write_args <- names(formals(write_csv_arrow))
+
+  unsupported_opts <- setdiff(
+    write_args,
+    union(arrow_write_opts, c("x", "sink"))
+  )
+
+  unsupported_passed_args <- names(unlist(passed_args[unsupported_opts]))
+
+  if (length(unsupported_passed_args)) {
+    stop(
+      "The following ",
+      ngettext(length(unsupported_passed_args), "argument is ", "arguments are "),
+      "not yet supported in Arrow: ",
+      oxford_paste(unsupported_passed_args),
+      call. = FALSE
+    )
+  }
```

Comment:
Yeah, I like this approach proposed by @paleolimbot here. It should also avoid any need for parsing the formals and comparing those.

You can also make a helper function for that [like we do with the reading side](readr_to_csv_parse_options)

Reply:
The link is broken, but I think I know what you mean. That was one of my other possible directions. Cool. Off to do some work and will update the PR.

Reply:
@jonkeane & @paleolimbot do we change the name of the write_csv_arrow() argument from sink to file? What about include_header to col_names?

Reply:
Yeah `sink` is a lot more than file path, and matches our other functions. How awkward would it be to support both sink and filename? It's a bit awkward, but it would make this match readr while also maintaining our arrow patterns.

Reply:
Done. @jonkeane please have a look to see if what I did makes sense. I think we could improve the arg description and maybe not `message()` about the whole `sink` `file` situation?